src/main/java/dukepkg/Parser.java

```diff
@@ -0,0 +1,310 @@
+            throw new FormatException("☹ OOPS!!! The " + arr[0] + " command should be " + arr[0] + " + task No.");
+        }
+        if (!arr[1].matches("(?<=\\s|^)\\d+(?=\\s|$)")){
+            throw new FormatException("☹ OOPS!!! The " + arr[0] + " command should be followed by a integer.");
+        }
+    }
+
+    /**
+     * Check whether the index used to modify existing tasks is out of range of the tasklist length.
+     *
+     * @param index      the index of the task in the list
+     * @param listLength the list length
+     * @throws FormatException if the index is out of range.
+     */
+    public static void validateModifyExistingTaskCommandIndex(int index, int listLength) throws FormatException {
+        if(index >= listLength || index < 0) {
+            throw new FormatException("☹ OOPS!!! The task No. you refer to is non-existent. Try another one.");
+        }
+    }
+
+    private void validateAddTaskCommandLength(String[] arr) throws FormatException {
+        if (arr.length == 1) {
+            throw new FormatException("☹ OOPS!!! The description of a " + arr[0] + " cannot be empty.");
+        }
+    }
+
+    /**
+     * Validate deadline format.
+     *
+     * @param arr the user input line.
+     * @throws FormatException thrown if deadline task format is wrong.
+     */
+    public static void validateDeadlineFormat(String[] arr) throws FormatException {
+        String[] ddl = arr[1].trim().split("/by", 2);
+        if(ddl.length < 2) {
+            throw new FormatException("☹ OOPS!!! Format of " + arr[0] + " should be deadline description /by time.");
+        } else if(arr[1].trim().matches("/by.*")) {
+            throw new FormatException("☹ oops!!! you forget to add description for the " + arr[0] + " command.");
+        } else if(arr[1].trim().matches(".*/by")) {
+            throw new FormatException("☹ oops!!! you forget to add time for the " + arr[0] + " command.");
+        }
+    }
+
+    /**
+     * Standardize time for the deadline task.
+     *
+     * @param arr the user input line.
+     * @return the Deadline task created after time format is standardized.
+     */
+    @SuppressWarnings("Annotator")
+    public static Task standardizeDeadlineTime(String[] arr) {
+        String[] ddl = arr[1].trim().split("/by", 2);
+        Task t;
+        String[] date_time = ddl[1].trim().split(" ");
+        //noinspection Annotator
+        if(date_time.length > 1 && date_time[0].matches("^((((([1-9])|(0[1-9])|(1\\d)|(2[0-8]))/(([1-9])|(0[1-9])|(1[0-2])))|((31/(((0[13578])|([13578]))|(1[02])))|((29|30)/(((0[1,3-9])|([1,3-9]))|(1[0-2])))))/((20[0-9][0-9]))|(((([1-9])|(0[1-9])|(1\\d)|(2[0-8]))/(([1-9])|(0[1-9])|(1[0-2])))|((31/(((0[13578])|([13578]))|(1[02])))|((29|30)/(((0[1,3-9])|([1,3-9]))|(1[0-2])))))/((19[0-9][0-9]))|(29/(02|2)/20(([02468][048])|([13579][26])))|(29/(02|2)/19(([02468][048])|([13579][26]))))$")  && date_time[1].matches("^(0[0-9]|1[0-9]|2[0-3]|[0-9])[0-5][0-9]$")) {
```

Comment:
If possible maybe you could break up this regex line into smaller chunks so you don't exceed the line limit. Though I'm think not many people know how to read regex, you could break it down so those who do know how to read it have an easier time trying to understand what you are doing.